docs/www/guide-wasm-elastic.md

```diff
@@ -1,261 +1,285 @@
-# Redpanda Wasm engine + Elasticsearch
+---
+title: Transforming data with WebAssembly (Wasm) - Elasticsearch tutorial
+order: 0
+---
+# Transforming data with WebAssembly (Wasm)
 
-The Wasm engine feature was introduced in version 21.2.1. In this guide, you'll
-learn to how create a wasm script, how publish record from Redpanda
-to Elasticsearch and how deploy a wasm script using `rpk wasm` command.
+When your application components send records to Redpanda,
+sometimes those records need to be transformed in thr process.
+Instead of adding another process to your application that consumes, transforms, and produces the records back to Redpanda,
+you can run Wasm transformations inside Redpanda.
 
-Requirements:
+In general, the steps for transforming data on-the-fly in Redpanda are:
 
-1. Redpanda version 21.2.1 or greater. You can check it by running `rpk version`.
-2. Elasticsearch, in this example we use the official [docker image](https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html)
+1. Create a Wasm script.
+2. Deploy the script using the `rpk wasm` command.
+3. Publish records to Redpanda.
 
->Note: this guide just work for linux enviroment
+In this tutorial, we'll use Redpanda installed on Linux to transform data and send it to Elasticsearch®.
+
+## Prerequisites
+
+For this tutorial you need:
+
+- Redpanda v21.2.1 or higher on [Linux](https://vectorized.io/docs/quick-start-linux)
+- Elasticsearch 7.10.2 or higher in a [docker image](https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html)
 
 ## Enable Wasm engine on Redpanda
-Add `coproc_enable: true` in your redpanda.yaml file, under the `redpanda` 
-section. You can do this manually or by running 
 
-```bash
-$ sudo rpk redpanda config set 'redpanda.enable_coproc' true
-If redpanda is running, please restart it for the changes to take effect.
-```
+By default the Wasm engine is disabled, so the first thing we need to do is enable it.
+
+To enable the Wasm engine:
+
+1. Set the `redpanda.enable_coproc` value in the Redpanda config to `true`:
+
+    ```bash
+    sudo rpk redpanda config set 'redpanda.enable_coproc' true
+    ```
 
-You will also need to set `developer_mode: true`. To do so, you can run
+    If redpanda is running, please restart it for the changes to take effect.
 
-```bash
-$ sudo rpk redpanda mode dev
-Writing 'dev' mode defaults to '/etc/redpanda/redpanda.yaml'
-```
+2. Set the `developer_mode` to `true`:
 
-Verify that your configuration is OK by checking your configuration 
-file (located in `/etc/redpanda/redpanda.yaml` by default):
+    ```bash
+    sudo rpk redpanda mode dev
+    ```
 
-```bash
-$ cat /etc/redpanda/redpanda.yaml | grep developer_mode              
```

Comment:
hmmm, we no longer need this. 

further mode we should use `rpk config set ...`